File: Codigos por competidor/Andres Espitia/Overflow.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
	long long numero1, numero2;
	char op;
	
	while(cin>>numero1>>op>>numero2){
		
		string n1 = to_string(numero1);
		string n2 = to_string(numero2);
		string imp = n1 + " " + op + " " + n2;
		
		cout<<imp<<endl;
		
		if(numero1>2147483647){
			cout<<"first number too big"<<endl;
		}
		
		if(numero2>2147483647){
			cout<<"second number too big"<<endl;
		}
		
		if((op=='+' && numero1+numero2>2147483647) || (op=='*' && numero1*numero2>2147483647)){
			cout<<"result too big"<<endl;
		}
	}
}

File: Resueltos_por_competidor/Andrés_Espitia/Overflow.java
package init2;





import java.math.BigInteger;
import java.util.Scanner;

public class Overflow {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		BigInteger num1, num2;
		String leer, dividir[], op;
		while(sc.hasNext()) {
			leer = sc.nextLine();
			dividir = leer.split(" ");
			
			
			num1 = new BigInteger(dividir[0]);
			op = dividir[1];
			num2 = new BigInteger(dividir[2]);
			
			System.out.println(leer);
			
			if(num1.compareTo(new BigInteger(Integer.toString(Integer.MAX_VALUE))) > 0) {
				System.out.println("first number too big");
			}
			
			if(num2.compareTo(new BigInteger(Integer.toString(Integer.MAX_VALUE))) > 0){
				System.out.println("second number too big");
			}
			
			if( (op.equals("+") && num1.add(num2).compareTo(new BigInteger(Integer.toString(Integer.MAX_VALUE))) > 0) || (op.equals("*") && num1.multiply(num2).compareTo(new BigInteger(Integer.toString(Integer.MAX_VALUE))) > 0)){
				System.out.println("result too big");
			}
		
		}
	}
}
